pkg/transport/process/util.go

```diff
@@ -0,0 +1,177 @@
+	}
+
+	marshaledRes, err := yaml.Marshal(res)
+	if err != nil {
+		return fmt.Errorf("unable to marshal resource: %w", err)
+	}
+
+	err = writeFileToTARArchive(resourceFile, bytes.NewReader(marshaledRes), outArchive)
+	if err != nil {
+		return fmt.Errorf("unable to write %s: %w", resourceFile, err)
+	}
+
+	if resourceBlobReader != nil {
+		err = writeFileToTARArchive(resourceBlobFile, resourceBlobReader, outArchive)
+		if err != nil {
+			return fmt.Errorf("unable to write %s: %w", resourceBlobFile, err)
+		}
+	}
+
+	return nil
+}
+
+func writeFileToTARArchive(filename string, contentReader io.Reader, outArchive *tar.Writer) error {
+	tempfile, err := ioutil.TempFile("", "")
+	if err != nil {
+		return fmt.Errorf("unable to create tempfile: %w", err)
+	}
+	defer tempfile.Close()
+
+	_, err = io.Copy(tempfile, contentReader)
+	if err != nil {
+		return fmt.Errorf("unable to write content to file: %w", err)
+	}
+
+	_, err = tempfile.Seek(0, 0)
+	if err != nil {
+		return fmt.Errorf("unable to seek to beginning of file: %w", err)
+	}
+
+	fstat, err := tempfile.Stat()
+	if err != nil {
+		return fmt.Errorf("unable to get file info: %w", err)
+	}
+
+	header := tar.Header{
+		Name:    filename,
+		Size:    fstat.Size(),
+		Mode:    int64(fstat.Mode()),
+		ModTime: time.Now(),
+	}
+
+	if err = outArchive.WriteHeader(&header); err != nil {
+		return fmt.Errorf("unable to write tar header: %w", err)
+	}
+
+	_, err = io.Copy(outArchive, tempfile)
+	if err != nil {
+		return fmt.Errorf("unable to write file to tar archive: %w", err)
+	}
+
+	return nil
+}
+
+// ReadTARArchive reads the component descriptor, resource and resource blob from a TAR archive.
+// The resource blob reader can be nil. If a non-nil value is returned, it must be closed by the caller.
+func ReadTARArchive(r *tar.Reader) (*cdv2.ComponentDescriptor, cdv2.Resource, io.ReadCloser, error) {
+	var cd *cdv2.ComponentDescriptor
+	var res cdv2.Resource
+	var f *os.File
+
+	for {
+		header, err := r.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, cdv2.Resource{}, nil, fmt.Errorf("unable to read tar header: %w", err)
+		}
+
+		switch header.Name {
+		case resourceFile:
+			res, err = readResource(r)
+			if err != nil {
+				return nil, cdv2.Resource{}, nil, fmt.Errorf("unable to read %s: %w", resourceFile, err)
+			}
+		case componentDescriptorFile:
+			cd, err = readComponentDescriptor(r)
+			if err != nil {
+				return nil, cdv2.Resource{}, nil, fmt.Errorf("unable to read %s: %w", componentDescriptorFile, err)
+			}
+		case resourceBlobFile:
+			f, err = ioutil.TempFile("", "")
+			if err != nil {
+				return nil, cdv2.Resource{}, nil, fmt.Errorf("unable to create tempfile: %w", err)
+			}
+			_, err = io.Copy(f, r)
+			if err != nil {
+				return nil, cdv2.Resource{}, nil, fmt.Errorf("unable to read %s: %w", resourceBlobFile, err)
+			}
+		}
+	}
+
+	if f != nil {
+		_, err := f.Seek(0, 0)
```

Comment:
not sure about the actual name, but we should use the constant of the golib (although its also just `0`)
```suggestion
		_, err := f.Seek(os.Offeset, 0)
```